display/widgets.go

```diff
@@ -23,48 +24,31 @@ var (
 	viewMode        int               = 1 // view type of instruction list (-1 -> raw, 1 -> op name)
 )
 
-// validateFileAndShowError checks if a file can be opened and if it's a .mif file.
-// If the file cannot be opened or is not a .mif file, it displays an error to the user.
-func validateFileAndShowError(f fyne.URIReadCloser, err error) {
+// fileSelect opens a file selection window and executes the callback after
+// opening the file and checking for errors.
+func fileSelect(callback func(io.ReadCloser)) {
+	fileName, err := fileDialog.File().Load()
 	if err != nil {
 		dialog.ShowError(err, window)
 		return
 	}
 
-	if f == nil {
-		dialog.ShowError(errors.New("could not open a file"), window)
+	file, err := os.OpenFile(fileName, os.O_RDONLY, 0)
+	if err != nil {
+		dialog.ShowError(err, window)
 		return
 	}
 
-	// Checks if the file has the .mif extension
-	if strings.ToLower(filepath.Ext(f.URI().Path())) != ".mif" {
```

Comment:
Talvez tirando esse check o erro que será gerado, quando um arquivo de extensão distinta de .mif (tipo o .asm) for carregado,  fique um pouco confuso pra um usuário mais leigo do simulador. Porque ele dá um erro na criação do parser só.